src/modules-load/modules-load.c

```diff
@@ -71,7 +90,27 @@ static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent
                 if (strchr(COMMENTS, *line))
                         continue;
 
-                k = module_load_and_warn(ctx, line, true);
+                RET_GATHER(r, send_module_to_load(sock, line));
+        }
+
+        return r;
+}
+
+static int run_prober(int sock) {
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        char buffer[NAME_MAX + 1];
+        ssize_t bytes_received;
+
+        int r = module_setup_context(&ctx);
+        if (r < 0)
+                log_error_errno(r, "Failed to initialize libkmod context: %m");
+
+        while ((bytes_received = recv(sock, buffer, sizeof(buffer), 0)) > 0) {
+                int k;
+
+                buffer[bytes_received] = '\0';
```

Comment:
this is a potential memory corruption. pass sizeof(buffer)-1 as size to recv() above to avoid it